packages/react/src/components/DesignLibrary/DesignLibrary.tsx

```diff
@@ -135,11 +139,13 @@ export const DesignLibrary = withLoadImportMap(({ loadImportMap }: DesignLibrary
         setRenderKey((k) => k + 1);
       });
 
-      const importMapEvent = getDesignLibraryImportMapEvent(rendering.uid, importMap);
+      const importMapEvent = getDesignLibraryImportMapEvent(uid, importMap);
       postToDesignLibrary(importMapEvent);
 
+      if (!propsState.fields || !propsState.params) return;
```

Comment:
Let's don't skip _componentProps_ even when fields or params are undefined. Instead, in [getDesignLibraryComponentPropsEvent](https://github.com/Sitecore/content-sdk/blob/chore/JSS-8592/packages/core/src/editing/codegen/preview.ts#L396-L397) you can make `fields` and `params` optional and set default parameters value as `{}`